### PR TITLE
Add shard command

### PR DESCRIPTION
**Pull request by TealNerd**
Sends a message "[BetterShards] You are currently in the shard 'ShardName'"

### CODE REVIEW

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			}
+			sender.sendMessage(ChatColor.GREEN + "You are currently in the shard '" + MercuryAPI.serverName() + "'");
+			return true;
+		} else {
+			if(!(sender.isOp() || (sender instanceof ConsoleCommandSender))) {
+				sender.sendMessage(ChatColor.RED + "You cannot look up the shard of another player!");
+				return true;
+			}
+			PlayerDetails details;
+			UUID player = UUID.fromString(args[0]);
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: You need to surround this in try catch.  Listen for IllegalArgumentException and then set player = null.

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import vg.civcraft.mc.mercury.MercuryAPI;
+import vg.civcraft.mc.mercury.PlayerDetails;
+
+public class GetShard extends PlayerCommand {
+
+	public GetShard(String name) {
+		super(name);
+		setIdentifier("shard");
+		setDescription("Gets the name of the shard you're in");
+		setUsage("/shard <name>");
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: Can you change name to player

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import vg.civcraft.mc.civmodcore.command.PlayerCommand;
+import vg.civcraft.mc.mercury.MercuryAPI;
+import vg.civcraft.mc.mercury.PlayerDetails;
+
+public class GetShard extends PlayerCommand {
+
+	public GetShard(String name) {
+		super(name);
+		setIdentifier("shard");
+		setDescription("Gets the name of the shard you're in");
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: Just add a period at the end of this sentence.

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		setUsage("/shard <name>");
+		setArguments(0, 1);
+	}
+	
+	@Override
+	public boolean execute(CommandSender sender, String[] args) {
+		if(args.length == 0) {
+			if(!(sender instanceof Player)) {
+				sender.sendMessage(ChatColor.RED + "Non-players need to specify a player to look up");
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: Just add a period at the end of this sentence.

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+				return true;
+			}
+			sender.sendMessage(ChatColor.GREEN + "You are currently in the shard '" + MercuryAPI.serverName() + "'");
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: Just add a period at the end of this sentence.

File: src/vg/civcraft/mc/bettershards/command/commands/GetShard.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+			if(player != null) {
+				details = MercuryAPI.getServerforAccount(player);
+			} else {
+				details = MercuryAPI.getServerforPlayer(args[0]);
+			}
+			if(details == null) {
+				sender.sendMessage(ChatColor.RED + "Player not found!");
+				return true;
+			} else {
+				sender.sendMessage(ChatColor.GREEN + args[0] + " is on the shard '" + details.getServerName() + "'");
```
Comment (rourke750) [inline src/vg/civcraft/mc/bettershards/command/commands/GetShard.java]: Just add a period at the end of this sentence.

**Comment by TealNerd** [comment]
@rourke750

**Comment by Maxopoly** [comment]
The functionality this was needed for was picked up by WorldInfo, so closing this.

**Pull request closed without merging**